File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l{0};
        int minlenght = 1000001;
        int currSum{0};

        for(int r = 0; r < nums.size(); r++){

            currSum += nums[r];

            while(currSum >= target){
                minlenght = min(minlenght, r - l + 1);
                currSum -= nums[l];
                l += 1;
            }
        }

        return (minlenght == 1000001)? 0 : minlenght;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        combinations = []
        combination = []

        def dfs(i, currSum):
            if currSum == target:
                combinations.append(combination.copy())
                return

            if currSum > target or i >= len(nums):
                return

            #including the num index
            combination.append(nums[i])
            dfs(i, currSum + nums[i])
            combination.pop()
            dfs(i+1, currSum)

        dfs(0,0)

        return combinations